frontend/src/Services/account-service.ts

```
@@ -88,32 +156,78 @@ const accountApi = {
 		}
 
 		// Only add StaffInfoRequest if all required fields are present
-		if (data.staffInfo && data.department) {
+		if (data.staffInfo && data.staffInfo.departmentId) {
 			transformedData.StaffInfoRequest = {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)